pyveda/training.py

```diff
@@ -240,70 +237,112 @@ class VedaCollection(BaseSet):
       Params:
           mlType (str): the type model this data may be used for training. One of 'classification', 'object detection', 'segmentation'
-          bbox (list): a list of geographic coordinate bounds for the data ([minx, miny, maxx, maxy])
-          classes (list): a list of names of the classes in the training data (ie buildings, cars, planes, trees)
-          source (str): Defaults to 'rda'. Currently anything but RDA imagery is treated as MapsAPI data.
-          shape (tuple): the shape of the imagery stored in the data. Used to enforce consistent shapes in the set.
-          dtype (str): the dtype of the imergy (ie int8, uint16, float32, etc)
+          tilesize (tuple): the shape of the imagery stored in the data. Used to enforce consistent shapes in the set.
+          partition (str):internally partition the contents into `train,validate,test` groups, in percentages. Default is `[100, 0, 0]`, all datapoints in the training group.
     """
-    def __init__(self, geojson, image, name, shape=None, mlType="classification", bbox=None, **kwargs):
+    def __init__(self, name, mlType="classification", tilesize=[256,256], partition=[100,0,0], **kwargs):
         assert mlType in valid_mltypes, "mlType {} not supported. Must be one of {}".format(mlType, valid_mltypes)
         super(VedaCollection, self).__init__()
-
-        self.image = image
-        self.geojson = None
-        self.bbox = bbox
-
-        if geojson is not None:
-            with NamedTemporaryFile(prefix="veda", suffix="json", delete=False) as temp:
-                with open(temp.name, 'w') as fh:
-                    geojson = json.loads(json.dumps(geojson)) # seriously wtf
-                    fh.write(json.dumps(geojson))
-                self.geojson = temp.name
-
-        self.id = kwargs.get('id', None)
-        self.links = kwargs.get('links')
-        if shape is None:
-            if self.image is not None:
-                self.shape = self.image.chunksize
-            else:
-                self.shape = (8,256,256)
-        else:
-            self.shape = tuple(map(int, shape))
+        #default to 0 bands until the first load
+        self.shape = [0] + list(tilesize)
+        self.partition = partition
         self.dtype = kwargs.get('dtype', None)
         self.percent_cached = kwargs.get('percent_cached', 0)
-        self.sensors = kwargs.get('sensors', [image.__class__.__name__])
+        self.sensors = kwargs.get('sensors', [])
+        self.rda_templates = kwargs.get('rda_templates', [])
         self._count = kwargs.get('count', 0)
         self._datapoints = None
+        self.id = kwargs.get('id', None)
+        self.links = kwargs.get('links')
 
         self.meta = {
             "name": name,
             "mlType": mlType,
             "public": kwargs.get("public", False),
             "partition": kwargs.get("partition", [100,0,0]),
-            "cache_type": kwargs.get("cache_type", "fetch"),
```

Comment:
Any reason to get rid of this? for now is nice for testing, though i think it now makes more sense as option in load.

Reply:
If it's an option then we should keep it. I editing out a bunch of stuff that looked old, that probably got prematurely cut.